<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="" xml:lang="">
<head>
  <meta charset="utf-8" />
  <meta name="generator" content="pandoc" />
  <meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
  <title>Class_projects</title>
  <style type="text/css">
      code{white-space: pre-wrap;}
      .smallcaps{font-variant: small-caps;}
      .line-block{white-space: pre-line;}
      .column{display: inline-block;}
  </style>
  <!--[if lt IE 9]>
    <script src="//cdnjs.cloudflare.com/ajax/libs/html5shiv/3.7.3/html5shiv-printshiv.min.js"></script>
  <![endif]-->
</head>
<body>
<p>Project Ideas for Class projects</p>
<ol type="1">
<li><p><em>MutMapper</em> - Identify transposon, EMS, or UV induced mutations in a strain as compared to a reference genome.</p></li>
<li><p><em>Transcriptomicon</em> - Develop a pipeline for processing raw RNAseq, assemnbling into transcripts, identifying ORFs, and extracting genes which have long UTRs, features.</p></li>
<li><p><em>Phylotantrum</em> - Build a pipeline to take a single gene sequence (protein) and identify homologs, align these, trim, and build gene trees. Refine tree or alignment to remove outliers and build a final gene alignment and Tree.</p></li>
<li><p><em>INDELible</em> - Taking population genomic resequencing data, identify insertion/deletion patterns across a genome. Are there non random distributions of indel size across a chromosome? Generate reports/interactive view of indel sizes and sequence preferences.</p></li>
</ol>
</body>
</html>
