<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="" xml:lang="">
<head>
  <meta charset="utf-8" />
  <meta name="generator" content="pandoc" />
  <meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
  <title>Homework_4</title>
  <style type="text/css">
      code{white-space: pre-wrap;}
      .smallcaps{font-variant: small-caps;}
      .line-block{white-space: pre-line;}
      .column{display: inline-block;}
  </style>
  <!--[if lt IE 9]>
    <script src="//cdnjs.cloudflare.com/ajax/libs/html5shiv/3.7.3/html5shiv-printshiv.min.js"></script>
  <![endif]-->
</head>
<body>
<ol type="1">
<li>Using python write a tool which generates a random subsampling tool for sequences. Given a FASTA sequence database file, which has 100,000 sequences, generate a new file which is a random subset these sequences selecting only 10% of them. Make this 10% an option in the program so it is easy to change to 20%, etc.</li>
</ol>
<p>See the script ‘rand_shuffle_seqs.py’ in the homework template</p>
<ol start="2" type="1">
<li>Run RNAseq analysis to compute the gene expression for two experiments. We will use data from this paper on light induction in bacterium Prochlorococcus</li>
</ol>
<p>See this paper:</p>
<ul>
<li><p>Thompson et al. PLoS One 2016; 11(10): e0165375 doi: <a href="http://dx.doi.org/10.1371/journal.pone.0165375">10.1371/journal.pone.0165375</a></p></li>
<li><p>Here is a view of the sequencing data. All data are on cluster in this folder <code>/bigdata/gen220/shared/projects/HW4/hw4_template</code></p></li>
</ul>
<p>Also you can get it from the web directly yourself. <a href="https://www.ebi.ac.uk/ena/data/view/PRJNA315575" class="uri">https://www.ebi.ac.uk/ena/data/view/PRJNA315575</a></p>
<p>The data for one timepoint: <a href="https://www.ebi.ac.uk/ena/data/view/SRR3234519&amp;display=html">0hrs light</a></p>
<p>Go and get the fastq files to process - see the <a href="https://github.com/biodataprog/hw4_hyphaltip_template/blob/master/download_bacteria_light_fastq.sh">download script</a></p>
<p>Light 0hr</p>
<ul>
<li><a href="ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR323/009/SRR3234519/SRR3234519_1.fastq.gz">0hr_1.fastq.gz</a></li>
<li><a href="ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR323/009/SRR3234519/SRR3234519_2.fastq.gz">0hr_2.fastq.gz</a></li>
</ul>
<p>The data for another timepoints: <a href="https://www.ebi.ac.uk/ena/data/view/SRR3234522&amp;display=html">4hrs light</a></p>
<p>The fastq files to process</p>
<ul>
<li><a href="ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR323/002/SRR3234522/SRR3234522_1.fastq.gz">4hr_1.fastq.gz</a></li>
<li><a href="ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR323/002/SRR3234522/SRR3234522_2.fastq.gz">4hr_2.fastq.gz</a></li>
</ul>
<p>Here is the Prochlorococcus marinus subsp. pastoris str. CCMP1986 genome: <a href="https://www.ncbi.nlm.nih.gov/assembly/GCF_000011465.1">GCF_000011465.1</a>. The RefSeq for this genome is at that link.</p>
<p>Can be obtained from this <a href="ftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCF/000/011/465/GCF_000011465.1_ASM1146v1/GCF_000011465.1_ASM1146v1_genomic.fna.gz">link</a></p>
<p>And the <a href="ftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCF/000/011/465/GCF_000011465.1_ASM1146v1/GCF_000011465.1_ASM1146v1_genomic.gff.gz">GFF</a></p>
<ul>
<li>Use Hisat2, samtools, and stringtie to generate a table of expression for these two experiments</li>
<li>Write a python script that will generate a report with 5 columns of data. Gene Name, Chromosome, Start, End, Strand, Gene Length, FPKM exp1, FPKM exp 2</li>
<li>Using R or other tools, make these plots
<ul>
<li>Plot gene expression from exp1 vs gene expression of exp 2</li>
<li>Plot gene length vs expr1 expression (FPKM). Is there a relationship?</li>
<li>Plot gene location vs expr2 (FPKM). Is there any relationship?</li>
</ul></li>
<li>print a list of genes which are &gt; 2 fold higher in the 4hr time point?</li>
</ul>
</body>
</html>
