<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="" xml:lang="">
<head>
  <meta charset="utf-8" />
  <meta name="generator" content="pandoc" />
  <meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
  <title>Homework_3</title>
  <style type="text/css">
      code{white-space: pre-wrap;}
      .smallcaps{font-variant: small-caps;}
      .line-block{white-space: pre-line;}
      .column{display: inline-block;}
  </style>
  <!--[if lt IE 9]>
    <script src="//cdnjs.cloudflare.com/ajax/libs/html5shiv/3.7.3/html5shiv-printshiv.min.js"></script>
  <![endif]-->
</head>
<body>
<h1 id="homework-3-advanced-python">Homework 3: Advanced Python</h1>
<ol type="1">
<li>Write a script <code>protein_freq.py</code> which will read in a protein Fasta format file <a href="https://biodataprog.github.io/programming-intro/data/Saccharomyces_cerevisiae.peps.fa.gz">Saccharomyces_cerevisiae.peps.fa</a>
<ol type="a">
<li>Print out the overall frequency (percentage) of each amino acid observed across all the sequences</li>
<li>Order this output sorted by the frequency of the amino acid, so the most frequent appear first.</li>
</ol></li>
<li><p>Using the Fasta files for the genomes ‘Ecoli_K-12.fasta’ and ‘B_subtilis_str_168.fasta’</p>
<ol type="a">
<li><p>Print out a table to compute frequency of all di-nucleotide combination (e.g. AA, AC, AG, AT, CA, CC, …).</p>
<p>Report should be tab delimited and look like</p>
<pre><code>Motif  Ecoli_K-12    B_subtilis_str_168
AA     7.28     9.85</code></pre></li>
</ol></li>
<li><p>Process a tabular BLAST report file ‘Ecoli-vs-Senterica.BLASTP.tab’ which has the following columns for a pairwise sequence alignment report</p></li>
</ol>
<p>QUERYNAME SUBJECTNAME PERCENTID LENGTHALN NUMMISMATCHES GAPOPEN QSTART QEND SSTART SEND EVALUE BITSCORE</p>
<p>Only print the lines which match the criteria:</p>
<ol type="a">
<li>SUBJECT ACCESSION between YP_008253351-YP_008253423</li>
<li>Percent Identity is &gt;= 25%</li>
</ol>
<p>Update the report to add 1 more columns after the existing ones.</p>
<ol type="a">
<li>Computed length of the query alignment using QSTART and QEND</li>
</ol>
<p>print out the new report on the STDOUT</p>
</body>
</html>
